### DISCUSSION TITLE
Help Discord bot

### DISCUSSION THREAD

**Discussion by MarsMars1** [Rapptz/discord.py#6650] [Q&A] [answered]
Hi! I know basically nothing about coding and for the last three hours i tried to make a discord bot. 

```py
import discord
client = discord.Client()
@client.event
def on_ready():
    print('We have logged in as {0.user}'.format(client))

@client.event
def on_message(message):
    if message.author == client.user:
        return
    if message.content.startswith('$hello'):
        message.channel.send('Hello!')
client.run('token')
```

this is what i wrote but it always gives me an error.
```
  File "<stdin>", line 7
    client.run('token')
    ^
SyntaxError: invalid syntax
```

i dont know how to fix it. please help.

**Comment by Rapptz** [comment, marked as answer]
There are a few issues here.

1. Events are meant to be `async def` rather than `def`. This should have errored earlier.
2. `message.channel.send` is a coroutine, so it needs `await` before it.

It's generally recommended to familiarise yourself with Python before starting to work on a Discord bot. Otherwise you'll end up struggling a lot when playing around with the library.

I am unsure where your SyntaxError is -- it looks syntactically valid. There might have been issue somewhere else, either with a stray character or potentially a transcribing issue. I'm not entirely sure.

**Reply by MarsMars1** [reply]
okay, thank you. I will look into python more.

**Comment by abe-101** [comment]
You probably did this but in case not
you need to change ('token') with the actual token
